types & grammar/ch2.md

```diff
@@ -1,214 +1,221 @@
 
-One very common way to make such a conversion is to borrow the `slice(..)` utility against the value:
+Uma forma comum de fazer tal conversão é usar a função `slice(..)` em um valor:
 
-```js
+
+```javascript
 function foo() {
-	var arr = Array.prototype.slice.call( arguments );
-	arr.push( "bam" );
-	console.log( arr );
+    var arr = Array.prototype.slice.call( arguments );
+    arr.push( "bam" );
+    console.log( arr );
 }
 
 foo( "bar", "baz" ); // ["bar","baz","bam"]
 ```
 
-If `slice()` is called without any other parameters, as it effectively is in the above snippet, the default values for its parameters have the effect of duplicating the `array` (or, in this case, `array`-like).
+Se `slice()` é chamada sem nenhum parâmetro, como no exemplo acima, os valores padrão de seus parâmetros tem o efeito de
+de duplicar o `array` (neste caso, uma estrutura semelhante a arrays).
 
-As of ES6, there's also a built-in utility called `Array.from(..)` that can do the same task:
+A partir da ES6, existe uma função padrão chamada `Array.from(..)` que pode fazer o mesmo efeito:
 
-```js
+
+```javascript
 ...
 var arr = Array.from( arguments );
 ...
 ```
 
-**Note:** `Array.from(..)` has several powerful capabilities, and will be covered in detail in the *ES6 & Beyond* title of this series.
+**Nota:** `Array.from(..)` possue várias capacidades poderosas, as quais iremos cobrir em detalhes em no livro _ES6 e Além_ desta série.
 
 ## Strings
 
-It's a very common belief that `string`s are essentially just `array`s of characters. While the implementation under the covers may or may not use `array`s, it's important to realize that JavaScript `string`s are really not the same as `array`s of characters. The similarity is mostly just skin-deep.
+É muito comum acreditar-se que uma `string` é essencialmente apenas um `array` de caracteres. Independentemente da implementação da linguagem usar ou não `array`s, é importante notar que `string`s, em Javascript, não são a mesma coisa que `array`s de caracteres. Sua similaridade é apenas superficial.
+Por exemplo, considere os dois valores abaixo:
 
 For example, let's consider these two values:
 
-```js
+```javascript
 var a = "foo";
 var b = ["f","o","o"];
 ```
 
-Strings do have a shallow resemblance to `array`s -- `array`-likes, as above -- for instance, both of them having a `length` property, an `indexOf(..)` method (`array` version only as of ES5), and a `concat(..)` method:
+Strings tem uma semelhança com `array` -- estruturas semelhantes a `array`s, no caso -- por exemplo, ambos tem uma propriedade `length` (tamanho, em português), um método `indexOf(..)` (`array`s apenas a partir da versão ES5) e um método `concat(..)` (concatenar, em português):
```

Comment:
Strings tem uma semelhança **superficial** com `array`s -- estruturas semelhantes a `array`s, **como acima** ~~no caso~~...

--

Colocar o significado de `length` e `concat` foi interessante, mas talvez colocar *tamanho* e *concatenar* em itálico ficaria melhor, o que acha?